### DISCUSSION TITLE
Restoring bilinear mesh values after update

### DISCUSSION THREAD

**Discussion by melvinmicallef** [Jyers/Marlin#142] [Q&A] [answered]
After updating firmware, it is recommend to perform an eeprom reset to default. This means that the matrix built from manual mesh bed levelling is lost. Is it possible to reapply the previous mesh (without actually re-levelling) using G29/M421 to set the 3x3 grid to the known values?

**Comment by dafi87** [comment]
The values aren't known anymore after you reset the eeprom.

**Comment by Jyers** [comment, marked as answer]
You could hypothetically take note of the mesh values and then re-enter them using G29 commands from a host. Take a look at this for more info on setting point values through GCODE https://marlinfw.org/docs/gcode/G029-abl-bilinear.html
